src/Arcus.Observability.Telemetry.Core/Extensions/IReadOnlyDictionaryExtensions.cs

```diff
@@ -0,0 +1,25 @@
+﻿using GuardNet;
+
+// ReSharper disable once CheckNamespace
+namespace System.Collections.Generic
+{
+    public static class IReadOnlyDictionaryExtensions
+    {
+        /// <summary>
+        ///     Provides value or default for a given dictionary entry
+        /// </summary>
+        /// <param name="dictionary">Dictionary containing data</param>
+        /// <param name="propertyKey">Key of the dictionary entry to return</param>
+        public static TValue GetValueOrDefault<TValue>(this IReadOnlyDictionary<string, TValue> dictionary, string propertyKey)
```

Comment:
Hmm, isn't `.TryGetValue` a method of the `IDictionary` type? Maybe we can make this more generilized?

Reply:
It doesn't give you the default value if it's not there so they are not identical?

Reply:
No, I mean use the `IDictionary` type in the signature:

```csharp
GetValueOrDefault<TValue>(this IDictionary dictionary...
```

Reply:
Ok can do that as well, will change this.

Reply:
Good point!

Reply:
Actually, `IReadOnlyDictionary` and `IDictionary` are not the same so I cannot use that

Reply:
Ok, it's fine.